src/main/java/mil/dds/anet/threads/AccountDeactivationWorker.java

```diff
@@ -0,0 +1,169 @@
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private final PersonDao dao;
+
+  private final List<Integer> daysTillEndOfTourWarnings;
+  private final List<String> ignoredDomains;
+
+  private final int warningIntervalInMs;
+
+  public AccountDeactivationWorker(AnetConfiguration config, PersonDao dao,
+      int warningIntervalInMs) {
+    this.dao = dao;
+
+    @SuppressWarnings("unchecked")
+    final List<Integer> daysTillWarning = (List<Integer>) config
+        .getDictionaryEntry("automaticallyInactivateUsers.emailRemindersDaysPrior");
+    this.daysTillEndOfTourWarnings = daysTillWarning;
+
+    @SuppressWarnings("unchecked")
+    List<String> domainsToIgnore = (List<String>) config
+        .getDictionaryEntry("automaticallyInactivateUsers.ignoredDomainNames");
+    this.ignoredDomains = domainsToIgnore.stream().map(x -> x.trim()).collect(Collectors.toList());
+
+    this.warningIntervalInMs = warningIntervalInMs;
+  }
+
+  @Override
+  public void run() {
+    logger.debug("Deactivation Warning Worker waking up to check for Future Account Deactivations");
+
+    // Check whether the application is configured to auto-check for account
+    // deactivation
+    if (this.daysTillEndOfTourWarnings == null) {
+      return;
+    }
+
+    try {
+      // Send warnings
+      Collections.sort(this.daysTillEndOfTourWarnings);
+      for (int i = 0; i < this.daysTillEndOfTourWarnings.size(); i++) {
+        int daysTillNextWarning = i == 0 ? -1
+            : this.daysTillEndOfTourWarnings.get(i) - this.daysTillEndOfTourWarnings.get(i - 1);
+        runInternal(this.daysTillEndOfTourWarnings.get(i), daysTillNextWarning);
+      }
+
+      // Deactivate accounts
+      runInternal(0, 0);
+    } catch (Throwable e) {
+      // Cannot let this thread die. Otherwise ANET will stop checking.
+      logger.error("Exception in run()", e);
+    }
+  }
+
+  private void runInternal(int daysUntilEndOfTour, int daysTillNextWarning) {
+    // Get a list of all people with a end of tour coming up
+    PersonSearchQuery query = new PersonSearchQuery();
+    query.setPageSize(Integer.MAX_VALUE);
+    Instant now = Instant.now().atZone(DaoUtils.getDefaultZoneId()).toInstant();
+    Instant warningDate = now.plus(daysUntilEndOfTour, ChronoUnit.DAYS);
+    query.setEndOfTourDateEnd(warningDate);
+    List<Person> persons = AnetObjectEngine.getInstance().getPersonDao().search(query).getList();
+    Instant nextReminder =
+        daysTillNextWarning < 0 ? null : now.plus(daysTillNextWarning, ChronoUnit.DAYS);
+
+    // Send emails to let users know their account will soon be deactivated
+    // or deactivate accounts that reach the end-of-tour date
+    for (Person p : persons) {
+
+      // Skip inactive ANET users or users from ignored domains
+      if (p.getStatus() == PersonStatus.INACTIVE
+          || this.ignoredDomains.stream().anyMatch(d -> p.getEmailAddress().matches(".*@" + d))) {
```

Comment:
Again, this matches more than you'd like, since the `.` in the dictionary's domain names now matches any character. Please look at `PersonResource::validateEmail` and extract the validation there into a common utility method used both here and there.